source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -158,9 +158,19 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
             throw mx::Exception("The element specified is not renderable.");
         }
 
+        bool asTexture = false;
+        if (parser.isFlagSet(kTextureFlag))
```

Comment:
To be orthogonal, maybe it's better to allow for type to be as a texture or as a shader. For instance we may want an unshaded graph to be used as a surface shader.

Reply:
Not sure I get the point. It's binary at the moment: either a texture or a  surface/shader. If the flag is set then the choice is determined by the flag, otherwise it's determined by the type of the element.

Reply:
In the else() condition you cannot set something to be used as surface shader if elementIsShader() returns that it's a texture.

Reply:
There's a negation there, on the assumption that if it's not a shader then it's a texture. If we ever have more than the two  options, we'll need to define an enum and map the command flag inputs to it.